File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def getRank(num):
    if num == 6:
        return 1
    if num == 5:
        return 2
    if num == 4:
        return 3
    if num == 3:
        return 4
    if num == 2:
        return 5
    return 6

def solution(lottos, win_nums):
    answer = []
    
    zero_count = lottos.count(0)
    max_value = 0
    min_value = 0
    
    for lotto in lottos:
        if lotto in win_nums:
            max_value += 1
            min_value += 1
    
    answer.append(getRank(max_value + zero_count))
    answer.append(getRank(min_value))
    
    return answer

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    const getRank = (number) => {
        if (number === 6) return 1;
        else if (number === 5) return 2;
        else if (number === 4) return 3;
        else if (number === 3) return 4;
        else if (number === 2) return 5;
        else return 6;
    }
    
    const currCount = lottos.filter((item) => win_nums.includes(item)).length;
    const zeroCount = lottos.filter((item) => item === 0).length;
    
    return [getRank(currCount + zeroCount), getRank(currCount)];
}